app_utils/sections/experiment.py

```diff
@@ -1596,19 +1649,23 @@ async def experiment_push_to_huggingface_dialog(q: Q, error: str = ""):
     elif q.args["experiment/display/push_to_huggingface_submit"]:
         await busy_dialog(
             q=q,
-            title="Exporting to HuggingFace ",
+            title="Exporting to HuggingFace",
             text="Model size can affect the export time significantly.",
         )
 
         experiment_path = q.client["experiment/display/experiment_path"]
         cfg, model, tokenizer = load_cfg_model_tokenizer(
-            experiment_path, merge=True, device="cpu"
+            experiment_path,
+            merge=True,
+            device=q.client["experiment/display/push_to_huggingface/device"],
         )
 
-        huggingface_hub.login(
-            q.client["experiment/display/push_to_huggingface/api_key"]
-        )
-        user_id = huggingface_hub.whoami()["name"]
+        user_id = q.client["experiment/display/push_to_huggingface/account_name"]
+        if user_id == "":
+            huggingface_hub.login(
+                q.client["experiment/display/push_to_huggingface/api_key"]
+            )
+            user_id = huggingface_hub.whoami()["name"]
```

Comment:
If no default token is provided (settings page) but a `user_id` in the textbox the user will never be logged in